aiohttp/resolver.py

```diff
@@ -32,8 +33,13 @@ async def resolve(
         )
 
         hosts = []
-        for family, _, proto, _, address in infos:
-            if family == socket.AF_INET6 and address[3]:  # type: ignore
+        for info in infos:
+            family, _, proto, _, address = info
+            # https://github.com/aio-libs/aiohttp/issues/5156
+            if family == socket.AF_INET6 and len(address) != 4:
```

Comment:
I think this heuristic is weak. It's better to set a proper flag for the `getaddrinfo()` call.